File: backend/src/main/java/com/careertuner/jobposting/domain/JobPosting.java
package com.careertuner.jobposting.domain;

import java.time.LocalDateTime;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class JobPosting {

    private Long id;
    private Long applicationCaseId;
    private Integer revision;
    private String originalText;
    private String uploadedFileUrl;
    private String extractedText;
    private String sourceType;
    private LocalDateTime createdAt;
}

File: frontend/src/features/applications/types/jobPosting.ts
import type { ApplicationSourceType } from "./applicationCase";

export interface JobPosting {
  id: number;
  applicationCaseId: number;
  revision: number;
  originalText: string | null;
  uploadedFileUrl: string | null;
  extractedText: string | null;
  sourceType: ApplicationSourceType;
  createdAt: string;
}

export interface JobPostingMetadata {
  companyName: string;
  jobTitle: string;
  postingDate: string | null;
  deadlineDate: string | null;
}

export interface JobPostingRequest {
  originalText?: string | null;
  uploadedFileUrl?: string | null;
  extractedText?: string | null;
  sourceType?: ApplicationSourceType;
}
